lib/utils.js

```diff
@@ -81,7 +81,10 @@ exports.deepEqual = function deepEqual(a, b) {
     return a.source === b.source &&
         a.ignoreCase === b.ignoreCase &&
         a.multiline === b.multiline &&
-        a.global === b.global;
+        a.global === b.global &&
+        a.dotAll === b.dotAll &&
+        a.unicode === b.unicode &&
+        a.sticky === b.sticky;
```

Comment:
It looks like we also need to check the `hasIndices` flag: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#advanced_searching_with_flags

Reply:
hasIndices is new in node16, but you are right.

Reply:
Done :)